app-modules/portal/src/Http/Controllers/KnowledgeManagementPortal/GetServiceRequestsController.php

```diff
@@ -86,4 +89,46 @@ public function __invoke(): JsonResponse
             )
         );
     }
+
+    public function show(ServiceRequest $serviceRequest): JsonResponse
+    {
+        $license = resolve(LicenseSettings::class);
+        $settings = resolve(PortalSettings::class);
+        $contact = auth('contact')->user();
+
+        $enabled = $license->data->addons->serviceManagement
+            && $settings->knowledge_management_portal_service_management
+            && $contact;
+
+        if (! $enabled) {
+            return response()->json();
+        }
+
+        return response()->json([
+            'serviceRequestDetails' => ServiceRequestDetailsData::from([
+                'id' => $serviceRequest->getKey(),
+                'title' => $serviceRequest->title,
+                'description' => $serviceRequest->close_details,
+                'serviceRequestNumber' => $serviceRequest->service_request_number,
+                'statusName' => $serviceRequest->status?->name,
+                'typeName' => $serviceRequest->priority?->type?->name,
+            ]),
+            'serviceRequestUpdates' => ServiceRequestUpdatesData::collection(
+                $serviceRequest
+                    ->serviceRequestUpdates()
+                    ->latest('created_at')
+                    ->get()
+                    ->map(function (ServiceRequestUpdate $serviceRequestUpdate) {
+                        return ServiceRequestUpdatesData::from([
+                            'id' => $serviceRequestUpdate->getKey(),
+                            'update' => $serviceRequestUpdate->update,
+                            'internal' => $serviceRequestUpdate->internal,
+                            'direction' => $serviceRequestUpdate->direction,
+                            'created_at' => $serviceRequestUpdate->created_at->format('m-d-Y g:i A'),
+                        ]);
+                    })
+                    ->toArray()
+            ),
```

Comment:
Like I mentioned in the DTO itself. I don't think a DTO is needed here as you are not moving this data through multiple layers.

Reply:
Done